src/coffee/helpers/sftp.coffee

```diff
@@ -257,13 +257,16 @@ class Sftp
           switch f.filename
             when '.', '..' then false
             else regex.test(f.filename)
-        Promise.all _.map filteredFiles, (f) => @stats(sftp, "#{remoteFolder}/#{f.filename}")
+        Promise.map filteredFiles, ((f) =>
+          @stats(sftp, "#{remoteFolder}/#{f.filename}")),
+          concurrency: 3
```

Comment:
limit seems reasonable but maybe you can define as parameter with default limit here and possibility to override it: https://github.com/sphereio/sphere-node-utils/pull/21/files#diff-c17bc85498662ec2f5d392798f13c410L250 and the caller